tests/sles4sap/publiccloud/qesap_ansible.pm

```diff
@@ -35,7 +36,48 @@ sub run {
     # skip ansible deployment in case of reusing infrastructure
     unless (get_var('QESAP_DEPLOYMENT_IMPORT')) {
         my @ret = qesap_execute(cmd => 'ansible', timeout => 3600, verbose => 1);
-        die("Ansible deploymend FAILED. Check 'qesap*' logs for details.") if $ret[0] > 0;
+        if ($ret[0]) {
+            if (qesap_file_find_string(file => $ret[1], search_string => 'Timed out waiting for last boot time check')) {
+                record_info('DETECTED ANSIBLE TIMEOUT ERROR');
+                $self->clean_up();
+                @ret = qesap_execute(
+                    cmd => 'terraform',
+                    verbose => 1,
+                    logname => 'qesap_terraform_retry.log.txt',
+                    timeout => 1800
+                );
+                die "'qesap.py terraform' return: $ret[0]" if ($ret[0]);
+                @ret = qesap_execute(
+                    cmd => 'ansible',
+                    verbose => 1,
+                    logname => 'qesap_ansible_retry.log.txt',
+                    timeout => 3600
+                );
+                if ($ret[0]) {
+                    qesap_cluster_logs();
+                    die "'qesap.py ansible' return: $ret[0]";
+                }
+                record_info('ANSIBLE RETRY PASS');
+            }
+            else {
+                qesap_cluster_logs();
+                die "'qesap.py ansible' return: $ret[0]";
+            }
+
+            my $instances = create_instance_data($provider);
```

Comment:
why this part is needed? Does it make sense to add a command to explain why?

Reply:
Good question. Because the retry (redeploy terraform/ansilbe) changed the instances, for example the ip address changed. see: https://openqa.suse.de/tests/12855001#step/deploy_qesap_ansible/117

Reply:
Sure I will add a common the explain.